front/pages/w/[wId]/builder/data-sources/new-public-url.tsx

```diff
@@ -0,0 +1,238 @@
+import { SubscriptionType } from "@dust-tt/types";
+import { APIError } from "@dust-tt/types";
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+import { useCallback, useEffect, useState } from "react";
+
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { AppLayoutSimpleSaveCancelTitle } from "@app/components/sparkle/AppLayoutTitle";
+import { subNavigationAssistants } from "@app/components/sparkle/navigation";
+import { getDataSources } from "@app/lib/api/data_sources";
+import { Authenticator, getSession, getUserFromSession } from "@app/lib/auth";
+import { classNames } from "@app/lib/utils";
+
+const { GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType | null;
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  dataSources: DataSourceType[];
+  gaTrackingId: string;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    context.params?.wId as string
+  );
+
+  const owner = auth.workspace();
+  const subscription = auth.subscription();
+  if (!owner || !subscription) {
+    return {
+      notFound: true,
+    };
+  }
+
+  if (!auth.isBuilder()) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const dataSources = await getDataSources(auth);
+
+  return {
+    props: {
+      user,
+      owner,
+      subscription,
+      dataSources,
+      gaTrackingId: GA_TRACKING_ID,
+    },
+  };
+};
+
+export default function DataSourceNew({
+  user,
+  owner,
+  subscription,
+  dataSources,
+  gaTrackingId,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const [isSaving, setIsSaving] = useState(false);
+  const [isEdited, setIsEdited] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+
+  const [dataSourceNameError, setDataSourceNameError] = useState("");
+  const [assistantDefaultSelected, setAssistantDefaultSelected] =
+    useState(true);
+
+  const [dataSourceUrl, setDataSourceUrl] = useState("");
+
+  const formValidation = useCallback(() => {
+    const urlRegex =
+      /^(https?:\/\/)?([\da-z.-]+)\.([a-z.]{2,6})([/\w .-]*)*\/?$/;
+
+    let edited = false;
+    let valid = true;
+
+    let exists = false;
+    dataSources.forEach((d) => {
+      if (d.name == dataSourceUrl) {
+        exists = true;
+      }
+    });
+    if (exists) {
+      setDataSourceNameError("A Folder with the same name already exists");
+      valid = false;
+    } else if (dataSourceUrl.length == 0) {
+      valid = false;
+      setDataSourceNameError("");
+    } else if (dataSourceUrl.startsWith("managed-")) {
+      setDataSourceNameError(
+        "DataSource name cannot start with the prefix `managed-`"
+      );
+      valid = false;
+      // eslint-disable-next-line no-useless-escape
```

Comment:
FMI where's the useless escape ? Why is it needed ? Should we kill this rule entirely ?